src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,57 +3,164 @@
+     * contains the block, street, unit and postalcode information.
+     */
+    public static boolean isValidAddress(String test) throws IllegalValueException {
+        String[] personDataSplit = test.split(",");
+        if (personDataSplit.length != LENGTH_OF_CORRECT_ADDRESS) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+        boolean isValidBlock = isValidBlock(personDataSplit[0]);
+        boolean isValidStreet = isValidStreet(personDataSplit[1]);
+        boolean isValidUnit = isValidUnit(personDataSplit[2]);
+        boolean isValidPostalcode = isValidPostalcode(personDataSplit[3]);
+        return isValidBlock && isValidStreet && isValidUnit && isValidPostalcode;
+    }
+
+    /**
+     * Returns true if a given string is a valid address PostalCode.
+     */
+    private static boolean isValidPostalcode(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.trim().matches(ADDRESS_POSTALCODE_VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns true if a given string is a valid address unit.
+     */
+    private static boolean isValidUnit(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.matches(ADDRESS_UNIT_VALIDATION_REGEX);
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid address street.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    private static boolean isValidStreet(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.matches(ADDRESS_STREET_VALIDATION_REGEX);
+
+    }
+
+    /**
+     * Returns true if a given string is a valid address block.
+     */
+    private static boolean isValidBlock(String string) {
+        String trimmedString = string.trim();
+        return trimmedString.trim().matches(ADDRESS_BLOCK_VALIDATION_REGEX);
     }
 
     @Override
     public String toString() {
-        return value;
+        return addressBlock.getBlock() + "," + addressStreet.getStreet() + "," + addressUnit.getUnit() + ","
+                + addressPostalcode.getPostalcode();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                        && this.addressBlock.equals(((Address) other).addressBlock)
+                        && this.addressStreet.equals(((Address) other).addressStreet)
+                        && this.addressUnit.equals(((Address) other).addressUnit)
+                        && this.addressPostalcode.equals(((Address) other).addressPostalcode)); // state
+                                                                                                // check
```

Comment:
Good indentation to make the code readable.
Though it can be improved.
Why is the comment at line 123 taking up an entire line?